autogen/agentchat/contrib/compressible_agent.py

```diff
@@ -0,0 +1,364 @@
+                    "TERMINATE": terminate the conversation ONLY when token count exceeds the max limit of current model.
+                        `trigger_count` is NOT used in this mode.
+                    "CUSTOMIZED": pass in a customized function to compress the messages.
+                - "compress_function" (Optional, callable, default to None): Must be provided when mode is "CUSTOMIZED".
+                    The function should takes a list of messages and returns a tuple of (is_compress_success: bool, compressed_messages: List[Dict]).
+                - "trigger_count" (Optional, float, int, default to 0.7): the threshold to trigger compression.
+                    If a float between (0, 1], it is the percentage of token used. if a int, it is the number of tokens used.
+                - "async" (Optional, bool, default to False): whether to compress asynchronously.
+                - "broadcast" (Optional, bool, default to True): whether to update the compressed message history to sender.
+                - "verbose" (Optional, bool, default to False): Whether to print the content before and after compression. Used when mode="COMPRESS".
+            **kwargs (dict): Please refer to other kwargs in
+                [ConversableAgent](../conversable_agent#__init__).
+        """
+        super().__init__(
+            name=name,
+            system_message=system_message,
+            is_termination_msg=is_termination_msg,
+            max_consecutive_auto_reply=max_consecutive_auto_reply,
+            human_input_mode=human_input_mode,
+            function_map=function_map,
+            code_execution_config=code_execution_config,
+            llm_config=llm_config,
+            default_auto_reply=default_auto_reply,
+        )
+
+        if compress_config:
+            if compress_config is True:
+                self.compress_config = {}
+            if not isinstance(compress_config, dict):
+                raise ValueError("compress_config must be a dict or True/False.")
+
+            assert compress_config.get("mode", "COMPRESS") in [
+                "COMPRESS",
+                "TERMINATE",
+                "CUSTOMIZED",
+            ], "compress_config['mode'] must be 'COMPRESS' or 'TERMINATE' or 'CUSTOMIZED'"
+
+            self.compress_config = {
+                "mode": "COMPRESS",
+                "compress_function": None,
+                "trigger_count": 0.7,
+                "async": False,
+                "broadcast": True,
+                "verbose": False,
+            }
+            self.compress_config.update(compress_config)
+            # convert trigger_count to int, default to 0.7
+            trigger_count = self.compress_config["trigger_count"]
+            if isinstance(trigger_count, float) and 0 < trigger_count <= 1:
+                self.compress_config["trigger_count"] = int(
+                    trigger_count * get_max_token_limit(self.llm_config["model"])
+                )
```

Comment:
What would happen here if trigger_count isn't a float, maybe this check should  happen earlier near here, so it can fail early and raise a ValueError if the expected type isn't found.
```python
. . .

if compress_config:
    if compress_config is True:
        self.compress_config = {}
    if not isinstance(compress_config, dict):
        raise ValueError("compress_config must be a dict 
  . . .
```

Reply:
Add checking to make sure it is an int or float. I think it is good to check it here.